### PR TITLE
Add an MOI wrapper

### PR DESCRIPTION
**Pull request by odow**
This PR wraps the MathProgBase nonlinear interface to AmplNLWriter. It provides `AmplNLWriter.Optimizer` with the same arguments as `AmplNLSolver`. 

<s>
Requires 
- [x] https://github.com/JuliaOpt/MathOptInterface.jl/pull/624
</s> 

Tests are now done using an MINLPTests.jl build stage on Travis.

cc @ccoffrin 

(Looking at certain unnamed people, please don't comment on how hacky this is.)

### CODE REVIEW

**Comment by odow** [comment]
> This wrapper seems to ignore linear and quadratic constraints and any MOI objective that might be set.

Yes, that is on my list as well. We just wanted a rough implementation that worked(ish) so LANL can start thinking about how to update their solvers without doing total re-writes.

**Comment by odow** [comment]
NLP testing is intended to be conducted via https://github.com/lanl-ansi/MINLPTests.jl

File: .travis.yml [hunk 1]
```diff
@@ -27,3 +23,11 @@ before_cache:
...
 after_success:
   - julia -e 'if VERSION >= v"0.7-"; using Pkg; else; cd(Pkg.dir("AmplNLWriter")); end; Pkg.add("Coverage"); using Coverage; Codecov.submit(process_folder())'
+jobs:
+  include:
+    - stage: "MINLPTests"
+      julia: 1.0
+      os: linux
+      script:
+        - julia --project=test/MINLPTests -e 'using Pkg; Pkg.instantiate(); Pkg.add(PackageSpec(path=pwd()))'
+        - julia --project=test/MINLPTests --color=yes test/MINLPTests/run_minlptests.jl
```
Comment (odow) [inline .travis.yml]: @ccoffrin we're testing AmplNLWriter with a  separate build stage on travis.

**Comment by odow** [comment]
This PR is now passing all linear and quadratic tests in MOI.Test, as well as all NLP and NLP-CVX tests in `MINLPTests.jl`, so I'm fairly happy. It substantially improves the rigorousness of the current tests.

Barring any objections, I'll merge this in a day or two.

 There are still some issues surrounding binary/discrete variables, e.g., #79, so we probably need to add Couenne/Bonmin to the `test/MINLPTests` project in order to run the NLP-MI tests, but that can wait for a future PR.

**Comment by ccoffrin** [comment]
What is the goal of adding Couenne/Bonmin to the `test/MINLPTests`?  To see if #79 is an upstream issue?

**Pull request merged**